compiler/env/OMRCPU.hpp

```diff
@@ -141,6 +137,16 @@ class CPU
     */
    bool supportsTransactionalMemoryInstructions() { return false; }
 
+   bool supportsFeature(uint32_t feature);
+   bool is(OMRProcessorArchitecture p) const;
+   bool isAtLeast(OMRProcessorArchitecture p) const;
+   bool isAtMost(OMRProcessorArchitecture p) const;
+   const OMRProcessorDesc & getProcessorDescription() const;
+
+protected:
+   OMRProcessorDesc _processorDescription;
+   uint32_t _featureMasks[OMRPORT_SYSINFO_FEATURES_SIZE];
```

Comment:
These need to be Doxygen documented. Also wondering if we can replace all the `is*` APIs by an implicit operator:

```
operator OMRProcessorArchitecture()
   {
   return _processorDescription.processor;
   }
```

Then the user can just use:

```
comp()->target().cpu == OMR_PROCESSOR_S390_Z15
```

For example. @Leonardo2718 / @dsouzai / @aviansie-ben thoughts on `is*` functions vs. operator?

Reply:
I'm not a fan of implicit type conversions. I would rather have `is()` methods or something like `comp()->target().cpu.version() == OMR_PROCESSOR_S390_Z15`.

Reply:
Adding an implicit conversion here would be incredibly dangerous IMO. Since `OMRProcessorArchitecture` is a C-style enum, this would also effectively add an implicit cast to `int`, which could cause confusing behaviour and would let nonsensical code like `comp()->target.cpu + comp()->target.cpu` compile. I agree with Leo that it's probably best to either use the `is*` methods that are already here or use another method to retrieve the `OMRProcessorArchitecture` explicitly.

Reply:
Thanks for the input. Only thing to address then is Doxygen documentation.